src/bin/commands/containers.rs

```diff
@@ -0,0 +1,499 @@
+        Ok(())
+    }
+}
+
+pub fn prune_volumes(
+    PruneVolumes {
+        verbose, execute, ..
+    }: PruneVolumes,
+    engine: &docker::Engine,
+) -> cross::Result<()> {
+    let mut command = docker::subcommand(engine, "volume");
+    command.args(&["prune", "--force"]);
+    if execute {
+        command.run(verbose, false).map_err(Into::into)
+    } else {
+        command.print_verbose(true);
+        Ok(())
+    }
+}
+
+fn get_package_info(
+    engine: &docker::Engine,
+    target: &str,
+    channel: Option<&str>,
+    docker_in_docker: bool,
+    verbose: bool,
+) -> cross::Result<(cross::Target, cross::CargoMetadata, docker::Directories)> {
+    let target_list = cross::target_list(false)?;
+    let target = cross::Target::from(target, &target_list);
+    let metadata = cross::cargo_metadata_with_args(None, None, verbose)?
+        .ok_or(eyre::eyre!("unable to get project metadata"))?;
+    let cwd = std::env::current_dir()?;
+    let host_meta = cross::version_meta()?;
+    let host = host_meta.host();
+    let sysroot = cross::get_sysroot(&host, &target, channel, verbose)?.1;
+    let dirs =
+        docker::Directories::create(engine, &metadata, &cwd, &sysroot, docker_in_docker, verbose)?;
+
+    Ok((target, metadata, dirs))
+}
+
+pub fn create_crate_volume(
+    CreateCrateVolume {
+        target,
+        docker_in_docker,
+        copy_registry,
+        verbose,
+        ..
+    }: CreateCrateVolume,
+    engine: &docker::Engine,
+    channel: Option<&str>,
+) -> cross::Result<()> {
+    let (target, metadata, dirs) =
+        get_package_info(engine, &target, channel, docker_in_docker, verbose)?;
+    let container = docker::remote::unique_container_identifier(&target, &metadata, &dirs)?;
+    let volume = format!("{container}-keep");
+
+    if docker::remote::volume_exists(engine, &volume, verbose)? {
+        eyre::bail!("error: volume {volume} already exists.");
+    }
+
+    docker::subcommand(engine, "volume")
+        .args(&["create", &volume])
+        .run_and_get_status(verbose, false)?;
+
+    // stop the container if it's already running
+    let state = docker::remote::container_state(engine, &container, verbose)?;
+    if !state.is_stopped() {
+        eprintln!("warning: container {container} was running.");
+        docker::remote::container_stop(engine, &container, verbose)?;
+    }
+    if state.exists() {
+        eprintln!("warning: container {container} was exited.");
+        docker::remote::container_rm(engine, &container, verbose)?;
+    }
+
+    // create a dummy running container to copy data over
+    let mount_prefix = Path::new("/cross");
+    let mut docker = docker::subcommand(engine, "run");
+    docker.args(&["--name", &container]);
+    docker.args(&["-v", &format!("{}:{}", volume, mount_prefix.display())]);
+    docker.arg("-d");
+    if atty::is(Stream::Stdin) && atty::is(Stream::Stdout) && atty::is(Stream::Stderr) {
+        docker.arg("-t");
+    }
+    docker.arg("ubuntu:16.04");
```

Comment:
does this have to be `ubuntu:16.04`? Is there a way to be smart about it, and use whatever `ubuntu` image is available.

Reply:
> does this have to be `ubuntu:16.04`? Is there a way to be smart about it, and use whatever `ubuntu` image is available.

Sure, this was just similar to what binfmt was doing in `register`. To avoid another docker call, we could probably just store our base image somewhere in the code, and just make a note to increment it (such as in `increment_versions`) whenever we upgrade it. Right now we almost always use 16.04, but soon that will be universally 20.04. Or we could do similar to what is done in `get_cross_containers`, where we run ` docker images --filter reference='ubuntu:*' --format "{{.Tag}}"`, and then select the latest version.

This would give output similar to:

```bash
$  docker images --filter reference='ubuntu:*' --format "{{.Tag}}"
22.04
latest
16.04
20.04
```

Reply:
might be something to consider for later.

this can be dismissed, I personally dislike the idea of an extra docker call

Reply:
Same, it's probably a good idea to store a base version though to avoid having multiple references to the same image scattered throughout the codebase.

Reply:
This now has a public constant with a note so we can update in future releases:

https://github.com/Alexhuszagh/cross/blob/d140c6dceee1fd294801a633f30a238ab858554a/src/docker/shared.rs#L17-L18

Reply:
perfect